polkadot/runtime/common/src/coretime/benchmarking.rs

```diff
@@ -0,0 +1,82 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! On demand assigner pallet benchmarking.
+
+#![cfg(feature = "runtime-benchmarks")]
+
+use super::{Pallet, *};
+use crate::coretime;
+use assigner_bulk::MAX_ASSIGNMENTS_PER_SCHEDULE;
+use frame_benchmarking::v2::*;
+//use frame_system::RawOrigin;
+//use xcm::latest::prelude::*;
+//use pallet_xcm::Origin;
+
+#[benchmarks]
+mod benchmarks {
+	use super::*;
+	#[benchmark]
+	fn assign_core(s: Linear<0, MAX_ASSIGNMENTS_PER_SCHEDULE>) {
+		// Setup
+		let caller: <T as frame_system::Config>::AccountId = whitelisted_caller();
+		// TODO: This value is copied from BROKER_ID in rococo runtime constants.
+		// Perhaps provide it in a more future proof way if worth the extra 
+		// dependency.
+		let broker_id: u32 = 1004;
```

Comment:
I see three options to resolve this:

1. Move the benchmark to the rococo runtime (and then we copy it to each chain) - not ideal. It is not that much code, but could still evolve.
2. Configure the [origin](https://github.com/paritytech/polkadot-sdk/blob/b9421f376f4db4b9fb18dcea295448776a04cf52/polkadot/runtime/common/src/coretime/mod.rs#L80) differently when benchmarks are enabled similar to what is done [here](https://github.com/paritytech/polkadot-sdk/blob/b9421f376f4db4b9fb18dcea295448776a04cf52/polkadot/runtime/rococo/src/lib.rs#L1286).
3. Avoid the origin check all together when doing benchmarks. Pretty sure it does not add any weight.

Reply:
The way it is now, will break because the broker chain will almost certainly have different ids on different networks.